source/sigv4.c

```diff
@@ -625,6 +627,20 @@ static SigV4Status_t lowercaseHexEncode( const SigV4String_t * pInputStr,
  */
 static size_t sizeNeededForCredentialScope( const SigV4Parameters_t * pSigV4Params );
 
+/**
+ * @brief Copy a string into a char * buffer.
+ * @note This function can be used to copy a string literal without
+ * MISRA warnings.
+ *
+ * @param[in] destination The buffer to write
+ * @param[in] source String to copy.
+ * @param[in] length Number of characters to copy.
+ * @return @p length
```

Comment:
I think `@return` needs to be elaborated a bit more

Reply:
Does it really? The third parameter is the return value, unchanged.

Reply:
If the third parameter is always returned, what is the point of returning it then?

Reply:
Everywhere this function is used, the buffer is incremented by the number of bytes written. Making it the return value means I can write
```c
pBuffer += copyString( pBuffer, str, length );
```
instead of
```c
copyString( pBuffer, str, length );
pBuffer += length;
```
which reduces complexity a little